app/services/create_or_update_location.rb

```diff
@@ -18,11 +18,7 @@ def build(params)
   end
 
   def update(params)
-    address = params.delete('address')
-    if address
-      address_params = build_address(address)
-      params.merge!(address_params)
-    end
+    params.merge!(build_address(params.delete('address')))
```

Comment:
There is a bunch of refactoring that doesn't really belong in this commit with the message it has. Can you pluck it out so that the guider stuff is in a single commit?